### PR TITLE
Remove forced unecessary asynchrony in flushing data to disk.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
A previous change i made made all flushing to disk *always* async.  This was to help avoid a situatoin where we might do synchronous flushing while under a lock.  That would be bad as it would mean the lock would potentially be held while doing IO, which could lead to a lot of backup with our threads.

However, this forced all the flushign to context switch over to another task *every time*.  I've been able to simplify this so that the thread that needs to flush can actually do the flushing itself, without having to move it to another task, and without having to hold a lock while doing so.  

--

I've also slowed down our bulk flush rate.  Testing two processes writing to sqlite on a heavily contended disk showed contention with the rate we were flushing before.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_WriteBatching.cs [hunk 1]
```diff
@@ -72,16 +73,38 @@ internal partial class SQLitePersistentStorage
...
+            // Note: this flushing is not cancellable.  We've already removed the
+            // writes from the write queue.  If we were not to write them out we
+            // would be losing data.
+            Debug.Assert(taskCompletionSource != null);
+
+            ProcessWriteQueue(connection, writesToProcess);
+
+            // Mark our TCS as completed.  Any other threads waiting on us will now be able
+            // to proceed.
+            taskCompletionSource.TrySetResult(0);
```
Comment (sharwell) [inline src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_WriteBatching.cs]: ❔ Should this be in a try/finally? Currently if `ProcessWriteQueue` ever throws all future calls to `FlushSpecificWritesAsync` for the same key would never complete.

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_WriteBatching.cs]: interesting point.  i think technically it shoudl be in a try/catch/finally.   will update.

**Comment by CyrusNajmabadi** [comment]
Telling @MattGertz about this.

This is just an efficiency improvement in the subsystem we use to persist data so it can be used by the host process and the OOP process.  A previous change of mine forced a thread context switch when that could be avoided.

**Pull request merged**